x-pack/filebeat/input/gcs/job.go

```diff
@@ -159,23 +146,18 @@ func (j *job) processAndPublishData(ctx context.Context, id string) error {
 }
 
 func (j *job) readJsonAndPublish(ctx context.Context, r io.Reader, id string) error {
-	var err error
-	r, err = j.addGzipDecoderIfNeeded(bufio.NewReader(r))
+	r, err := j.addGzipDecoderIfNeeded(bufio.NewReader(r))
 	if err != nil {
 		return fmt.Errorf("failed to add gzip decoder to object: %s, with error: %w", j.object.Name, err)
 	}
 
-	// if offset == 0, then this is a new stream which has not been processed previously
-	if j.offset == 0 {
-		r, j.isRootArray, err = evaluateJSON(bufio.NewReader(r))
-		if err != nil {
-			return fmt.Errorf("failed to evaluate json for object: %s, with error: %w", j.object.Name, err)
-		}
-		if j.isRootArray {
-			j.state.setRootArray(j.object.Name)
-		}
+	r, j.isRootArray, err = evaluateJSON(bufio.NewReader(r))
+	if err != nil {
+		return fmt.Errorf("failed to evaluate json for object: %s, with error: %w", j.object.Name, err)
 	}
+
 	dec := json.NewDecoder(r)
+	dec.UseNumber()
```

Comment:
Nice to know why [UseNumber](https://pkg.go.dev/encoding/json#Decoder.UseNumber) here?

Reply:
I'm not convinced a comment is needed since this is an idiom.